src/clusterfuzz/_internal/build_management/build_manager.py

```diff
@@ -868,9 +897,13 @@ def _unpack_custom_build(self):
         os.makedirs(directory)
       gcs_path = f'/{custom_builds_bucket}/{self.custom_binary_key}'
       storage.copy_file_from(gcs_path, build_local_archive)
-    elif not blobs.read_blob_to_disk(self.custom_binary_key,
+    else:
+      gcs_path = blobs.get_gcs_path(self.custom_binary_key)
+      if not blobs.read_blob_to_disk(self.custom_binary_key,
                                      build_local_archive):
-      return False
+        return False
+
+    self._emmit_build_age_metric(gcs_path)
```

Comment:
I don't think this function (`_unpack_build`) is the correct place to track this metric. 

This gets called to legitimately unpack older builds in e.g. regression_task where we are performing bisection to find the first build that introduces a crash. 

We need to do this only in the `setup_trunk_build` path, which is the path for setting up the **latest** build.

Reply:
If we do it on setup_trunk_build though, the code path for setup_custom_binary will never get hit, and we will not emmit metrics for chrome fuzzers, right?

setup_trunk_build calls setup_regular_build under the hood:

https://github.com/google/clusterfuzz/blob/24f976051fbe6b02660cc02f9d5c87986b01c981/src/clusterfuzz/_internal/build_management/build_manager.py#L1158

And if we only allow for classes that inherit from RegularBuild, we will not instrument SymbolizedBuild nor CustomBuild, which are legitimate use cases for us, right?

Having said that, if the concern is instrumenting other tasks that legitimately will download old builds, we can use the "task_name" label to filter out for only fuzz task. Wdyt?

Reply:
We don't need to track this at all for custom builds. Custom builds are manually uploaded builds with no expectation they'll be updated regularly. They're typically used for one-off experiments. 

For `SymbolizedBuild` etc -- that is not used for fuzzing.

I don't think we should tie the task name to whether or not we want to alert on it or not. That's tying an assumption that's a bit far removed from the actual thing we're checking -- "when we pull what we expect to be the latest builds, are they actually kept up to date?".

re `setup_trunk_build` -- I don't think we should (or can) call emit there, because the full zip url is not constructed there -- it's `setup_regular_build` that has this. We just need to limit the cases when we emit to when we know `setup_regular_build` is setting up the latest. i.e. we might need to just add a boolean there that is set when called from `setup_trunk_build`. Perhaps there's something more elegant here that achieves the same.

Reply:
Thanks for the clarification! Will implement that.

Reply:
Another not on this, it is possible that setup_regular_build gets invoked with the latest revision, coming from some code path other than setup_trunk_build.

setup_regular_build is indeed the correct place to emit this metric, IFF the revision in question is the latest.